src/base/static/state/ducks/map.js

```diff
@@ -1,311 +1,581 @@
+                ),
+              },
+            },
+          },
         },
       };
 
-      // If a previous basemap was already visible, switch it off.
-      if (state.visibleBasemapId && state.visibleBasemapId !== action.payload) {
-        newStatuses[state.visibleBasemapId] = {
-          ...state.layerStatuses[state.visibleBasemapId],
-          isVisible: false,
-        };
-      }
-
+    case REMOVE_FOCUSED_GEOJSON_FEATURES:
       return {
         ...state,
-        visibleBasemapId: action.payload,
-        layerStatuses: {
-          ...state.layerStatuses,
-          ...newStatuses,
+        style: {
+          ...state.style,
+          sources: {
+            ...state.style.sources,
+            "__mapseed-focused-source__": {
+              type: "geojson",
+              data: {
+                type: "FeatureCollection",
+                features: [],
+              },
+            },
+          },
         },
       };
-    case INIT_LAYER:
-      newStatuses = action.payload.reduce((memo, loadStatus) => {
-        return {
-          ...memo,
-          [loadStatus.id]: {
-            ...loadStatus,
-          },
-        };
-      }, {});
-
+    case UPDATE_FOCUSED_GEOJSON_FEATURES:
       return {
         ...state,
-        layerStatuses: {
-          ...state.layerStatuses,
-          ...newStatuses,
+        style: {
+          ...state.style,
+          sources: {
+            ...state.style.sources,
+            "__mapseed-focused-source__": {
+              type: "geojson",
+              data: {
+                type: "FeatureCollection",
+                features: action.payload,
+              },
+            },
+          },
         },
       };
-    case UPDATE_LAYERS_VISIBLE:
-      newStatuses = action.payload.reduce((memo, layerId) => {
-        return {
-          ...memo,
-          [layerId]: {
-            ...state.layerStatuses[layerId],
-            isVisible: true,
-          },
-        };
-      }, {});
-
+    case UPDATE_VIEWPORT:
       return {
         ...state,
-        layerStatuses: {
-          ...state.layerStatuses,
-          ...newStatuses,
+        viewport: {
+          ...state.viewport,
+          ...action.payload.viewport,
+          // NOTE: This is a fix for an apparent bug in react-map-gl.
+          // See: https://github.com/uber/react-map-gl/issues/630
+          bearing: isNaN(action.payload.viewport.bearing)
+            ? state.viewport.bearing
+            : action.payload.viewport.bearing,
+          // These checks support a "scroll zoom around center" feature (in
+          // which a zoom of the map will not change the centerpoint) that is
+          // not exposed by react-map-gl. These checks are pretty convoluted,
+          // though, so it would be great if react-map-gl could just
+          // incorporate the scroll zoom around center option natively.
+          // See: https://github.com/uber/react-map-gl/issues/515
+          latitude:
```

Comment:
These `latitude` and `longitude` updates are a little hacky. I experimented with some other approaches to maintain parity with the "scroll around center" functionality but this was the best I could come up with.

Ideally `react-map-gl` would support zoom around center out of the box. Maybe that will come in a future release, or failing that maybe we could fork `react-map-gl` down the road and add a more robust scroll around center feature.